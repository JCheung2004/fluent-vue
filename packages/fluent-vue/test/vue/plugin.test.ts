import { createLocalVue, mount } from '@vue/test-utils'

import { FluentBundle, FluentResource } from '@fluent/bundle'
import ftl from '@fluent/dedent'

import FluentVue from '../../src'

describe('vue integration', () => {
  const localVue = createLocalVue()
  localVue.use(FluentVue)

  const bundle = new FluentBundle('en-US')

  bundle.addResource(
    new FluentResource(ftl`
    message = Hello, { $name }!
    sub-message = Hi, { $name }
    `)
  )

  const fluent = new FluentVue({
    bundles: [bundle]
  })

  const options = {
    fluent,
    localVue
  }

  it('translates messages in a component', () => {
    // Arrange
    bundle.addResource(
      new FluentResource(ftl`
      message = Hello, { $name }!
      `)
    )

    const component = {
      data: () => ({
        name: 'John'
      }),
      template: "<div>{{ $t('message', { name }) }}</div>"
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })

  it('translates messages in sub-component', () => {
    // Arrange
    const child = {
      data: () => ({
        name: 'Alice'
      }),
      template: "<div>{{ $t('sub-message', { name }) }}</div>"
    }

    const component = {
      components: {
        child
      },
      data: () => ({
        name: 'John'
      }),
      template: `
        <div>
          {{ $t('message', { name }) }}
          <child />
        </div>
      `
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })

  it('clears instance on component destroy', () => {
    // Arrange
    const component = {
      data: () => ({
        name: 'john'
      }),
      template: "<div>{{ $t('message', { name }) }}</div>"
    }

    const mounted = mount(component, options)

    // Act
    mounted.destroy()

    // Assert
    expect(mounted.vm.$fluent).not.toBeUndefined()

    localVue.nextTick(() => {
      expect(mounted.vm.$fluent).toBeUndefined()
    })
  })

  it('does not try to clear if not initialized', () => {
    // Arrange
    const component = {
      template: '<div>Just text</div>'
    }

    const mounted = mount(component, {
      localVue
    })

    // Act
    mounted.destroy()

    // Assert
    expect(mounted.vm.$fluent).toBeUndefined()

    localVue.nextTick(() => {
      expect(mounted.vm.$fluent).toBeUndefined()
    })
  })
})
