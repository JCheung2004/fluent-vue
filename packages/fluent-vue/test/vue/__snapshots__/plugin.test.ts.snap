// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue integration translates messages in a component 1`] = `<div>Hello, ⁨John⁩!</div>`;

exports[`vue integration translates messages in sub-component 1`] = `
<div>
  Hello, ⁨John⁩!
  <div>Hi, ⁨Alice⁩</div>
</div>
`;
